src/Graphics/Vty/Platform/Windows/WindowsInterfaces.hs

```diff
@@ -45,36 +45,38 @@ readBuf' eventChannel inputEventPtr handle bufferPtr maxInputRecords = do
         handleInputEvent (offset, mSurrogateVal) inputEvent = do
             case inputEvent of
                 KeyEventRecordU (KeyEventRecordC isKeyDown _ _ _ cwChar _) -> do
-                    if isKeyDown
+                    -- Process the character if this is a 'key down' event,
+                    -- AND the char is not NULL
+                    if isKeyDown && cwChar /= 0
                     then processCWChar (offset, mSurrogateVal) $ fromEnum cwChar
                     else return (offset, Nothing)
                 WindowBufferSizeRecordU (WindowBufferSizeRecordC (COORD x y)) -> do
                     let resize = EvResize (fromIntegral x) (fromIntegral y)
                     atomically $ writeTChan eventChannel (InputEvent resize)
                     return (offset, Nothing)
-                -- Drop focus events for now, since we should receive equivalent virtual
-                -- terminal sequences when focus mode is enabled.
-                FocusEventRecordU (FocusEventRecordC _) -> return (offset, Nothing)
                 _ -> return (offset, Nothing)
 
         processCWChar :: (Int, Maybe Int) -> Int -> IO (Int, Maybe Int)
         processCWChar (offset, Nothing) charVal = do
             if isSurrogate charVal
             then return (offset, Just charVal)
-            else do
-                let utf8Char = encodeChar $ toEnum charVal
-                writeToBuffer utf8Char offset
-                return (offset + length utf8Char, Nothing)
+            else encodeAndWriteToBuf offset charVal
+            where
+                isSurrogate :: Int -> Bool
+                isSurrogate c = 0xD800 <= c && c < 0xDC00
         processCWChar (offset, Just surogateVal) charVal = do
-            let utf8Char = encodeChar $ toEnum $ (((surogateVal .&. 0x3FF) `shiftL` 10) .|. (charVal .&. 0x3FF)) + 0x10000
+            let utf8Char = toEnum $ (((surogateVal .&. 0x3FF) `shiftL` 10) .|. (charVal .&. 0x3FF)) + 0x10000
```

Comment:
This is wrong... Using toEnum twice? I suppose it doesn't break anything at least, but it's bad.